compiler/rustc_span/src/lib.rs

```diff
@@ -670,6 +691,7 @@ impl Span {
             cmp::min(span_data.lo, end_data.lo),
             cmp::max(span_data.hi, end_data.hi),
             if span_data.ctxt == SyntaxContext::root() { end_data.ctxt } else { span_data.ctxt },
+            if span_data.parent == end_data.parent { span_data.parent } else { None },
```

Comment:
Do I understand correctly that setting `parent` to `None` effectively means "giving up" on the incremental optimization, but otherwise doesn't affect correctness in any way?

Reply:
Yes.